airflow/www/views.py

```diff
@@ -3724,8 +3724,38 @@ def conf(self):
         raw = request.args.get('raw') == "true"
         title = "Airflow Configuration"
         subtitle = AIRFLOW_CONFIG
+
+        expose_config = conf.get('webserver', 'expose_config')
+
         # Don't show config when expose_config variable is False in airflow config
-        if conf.getboolean("webserver", "expose_config"):
+        # Don't show sensitive config values if expose_config variable is 'non-sensitive-only'
+        # in airflow config
+        if expose_config.lower() == 'non-sensitive-only':
+            from airflow.configuration import SENSITIVE_CONFIG_VALUES
+
+            with open(AIRFLOW_CONFIG) as file:
+                config = file.readlines()
+                for line in config:
+                    for _, key in SENSITIVE_CONFIG_VALUES:
```

Comment:
This is making an assumption that the name is unique amongst all sections. I don't think it's a problem (better to redact too much than not enough) but there is a possibility of this catching other settings.

As I said, not a problem per se, but I would add a comment to that effect.

Reply:
Added a comment on this regard